consensus/XDPoS/engines/engine_v2/engine.go

```diff
@@ -376,9 +459,18 @@ func (x *XDPoS_v2) onCountdownTimeout(time time.Time) error {
 }
 
 func (x *XDPoS_v2) broadcastToBftChannel(msg interface{}) {
-	x.BroadcastCh <- msg
+	go func() {
+		x.BroadcastCh <- msg
+	}()
 }
 
 func (x *XDPoS_v2) getCurrentRoundMasterNodes() []common.Address {
 	return []common.Address{}
 }
+
+func (x *XDPoS_v2) getSyncInfo() utils.SyncInfo {
+	return utils.SyncInfo{
+		HighestQuorumCert:  *x.highestQuorumCert,
+		HighestTimeoutCert: *x.highestTimeoutCert,
```

Comment:
what if they are `nil`?

Reply:
they should not be nil. they are initilised with 
```highestTimeoutCert: &utils.TimeoutCert{},
		highestQuorumCert:  &utils.QuorumCert{},
```
in the `New` method